connection_id.go

```diff
@@ -23,6 +23,6 @@ func RandomCIDGenerator(size int) func() []byte {
 // necessary.
 func OnlySendCIDGenerator() func() []byte {
```

Comment:
Instead of this, I think it's better to be ultra explicit and use a second return value of a boolean to indicate, there is a CID. Do you think it makes sense to do something like this now or just make the fix and consider it later?

Reply:
Hmm, I don't love the idea of expanding the signature just for a single scenario when we are allowing folks to supply their own generators. We say that the presence of a connection ID generator enables CID support, which would then only be true if the generator returned `true`. A potential alternative would be to evaluate whether the generated CID is `nil` in flight 1, then set the local connection ID to `[]byte{}` instead to ensure we properly set it in the second client hello. That would make returning `nil` or `[]byte{}` from a generator have the same effect, which was the original intention.
https://github.com/pion/dtls/blob/83b125485ec0277011ba97d26c2c5578ea96d8a7/flight1handler.go#L124-127

Thoughts?

Reply:
I like that idea!

Reply:
Otherwise there's more onus on other impls of this to correctly distinguish from nil/[]byte{} which is uncommon from what I've seen with slice usage

Reply:
Awesome! Update incoming 👍🏻